library/goptions.mli

```diff
@@ -192,6 +192,9 @@ type table_value =
   | StringRefValue of string
   | QualidRefValue of Libnames.qualid
 
+(** [get_option_value key] returns [None] if option with name [key] was not found. *)
+val get_option_value : option_name -> option_value option
```

Comment:
Maybe this should be 
```suggestion
val get_option_value : option_name -> (unit -> option_value) option
```
instead? Then the metacoq code would look like
~~~ocaml
let timing_opt =
  let open Goptions in
  let key = ... in
  match get_option_value key with
  | Some get -> get
  | None -> declare_bool_and_ref ... (* no () argument *)
~~~

Reply:
I don't have a strong opinion, and can easily change this. But doesn't the current API look more natural than the one you suggest, if you forget about the specific Metacoq use case?

Reply:
I feel the metacoq usage is a bit of a phase separation violation, even if it's a benign one. The `let _ = timing_opt () (* just for init *)` is both ugly and fetches a Interp-stage value at plugin link time.